### ISSUE TITLE
set curl options globally

### ISSUE THREAD

**Issue by sckott** [ropensci/crul#48]
main use case is when using `crul` inside  a package, becomes esp. important when e.g., a pkg `foo` uses `crul`, then another pkg `bar` uses `foo`, which makes likelihood of arg. collisions more likely

**Comment by sckott** [comment]
@maelle can you give it a try when you get a chance? if so, install like `remotes::install_github("ropensci/crul@set-options")` and then see the examples in `?crul-options`

note that the options set with these new functions are only picked up on function intialization, so if you set new options you have to remake the client object. let me know if that seems good or not.

**Comment by sckott** [comment]
also @dirkschumacher and @mikkelkrogsholm - see above comment, any thoughts on how these functions are implemented? let me know if any questions/problems

**Comment by maelle** [comment]
Does this mean an user could set the agent in their .Rprofile for instance and thus overrides all possible `agent` arguments? (which would be good I guess cf https://twitter.com/hrbrmstr/status/986339163833339904)

**Comment by sckott** [comment]
doesn't support using a file, but I did think about that. 

I looked around at other http clients and didn't see any examples of using a file. But doesn't mean we can't do it. I do worry a bit about users setting these things in a file and then forgetting about them, and then wondering what's going wrong later on. I guess one could simply give a message or so reminding the user that they are using options from a file.
